docs/usage/validators.md

```diff
@@ -419,15 +292,120 @@ except ValidationError as e:
     """
 ```
 
-As with field validators, root validators can have `pre=True`, in which case they're called before field
-validation occurs (and are provided with the raw input data), or `pre=False` (the default), in which case
-they're called after field validation.
+Model validators can be `mode='before'` or `mode='after'` and `mode='wrap'`.
+
+Before model validators are passed the raw input which is often a `dict[str, Any]` but could also be an instance of the model itself (e.g. if `UserModel.model_validate(UserModel.construct(...))` is called) or anything else since you can pass arbitrary objects into `model_validate`.
+Because of this before validators are extremely flexible and powerful but can be cumbersome and error prone to implement.
+Before model validators should be class methods.
+The first argument should be `cls` (and we also recommend you use `@classmethod` below `@model_validator` for proper type checking), the second argument will be the input (you should generally type it as `Any` and use `isinstance` to narrow the type) and the third argument (if present) will be a `pydantic.ValidationInfo`.
+
+After validators are instance methods and always receive an instance of the model as the first argument.
+You should not use `(cls, ModelType)` as the signature, instead just use `(self)` and let type checkers infer the type of `self` for you.
+Since these are fully type safe they are often easier to implement than before validators.
+If any field fails to validate after validators will not be called.
```

Comment:
Another good example that I think could be hard for a newcomer to make sense of right away. If you don't immediately realize that "after validators" need to be grouped together as a concept, it is easy to read one word at a time and think that it's about to explain to me that something might happen before a field fails to validate that would affect something later, and then think "was there a typo/grammar error here?".

I think the following prevents that misread:
```suggestion
If any field fails to validate, `mode='after'` validators will not be called.
```